packages/patternfly-4/react-core/src/experimental/components/DataToolbar/DataToolbarContent.tsx

```diff
@@ -12,17 +13,107 @@ export interface DataToolbarContentProps extends React.HTMLProps<HTMLDivElement>
   breakpointMods?: DataToolbarBreakpointMod[];
   /** Content to be rendered as children of the content row */
   children?: React.ReactNode;
+  /** Flag indicating if a Data toolbar toggle group's expandable content is expanded */
+  isExpanded?: boolean;
+  /** A callback for setting the isExpanded flag */
+  toggleIsExpanded?: () => void;
+  /** Optional callback for clearing all filters in the toolbar */
+  clearAllFilters?: () => void;
+  /** Flag indicating that the Clear all filters button should be visible */
+  showClearFiltersButton?: boolean;
 }
 
-export const DataToolbarContent: React.FunctionComponent<DataToolbarContentProps> = ({
-  className,
-  children,
-  breakpointMods = [] as DataToolbarBreakpointMod[],
-  ...props
-}: DataToolbarContentProps) => {
-  return (
-    <div className={css(styles.dataToolbarContent, formatBreakpointMods(breakpointMods, styles), className)} {...props}>
-      {children}
-    </div>
-  );
-};
+export interface DataToolbarContentState {
+  /** Flag indicating the if the expandable content's expanded state is consumer managed or not */
+  isConsumerManagedToggleGroup: boolean;
+  /** Flag indicating if the component managed state has expanded content or not */
+  componentManagedIsExpanded: boolean;
```

Comment:
The `isConsumerManagedToggleGroup` and `componentManagedIsExpanded` names are a little confusing to me. 

Perhaps something more like `isToggleManaged` would suffice?

Reply:
I'm struggling with the `componentManagedIsExpanded` name. Couldn't a user use `isExpanded` directly, when `isConsumerManagedToggleGroup` is in use? 

Could this be simplified at all?

Reply:
I will change the names of the `isConsumerManagedToggleGroup and componentManagedIsExpanded` to try and clear it up some. 

Basically, I allow the consumer to control the expanded state of the DataToolbarContentGroup by passing `isExpanded` and `toggleIsExpanded` as props. If they are not provided by the consumer, then both are managed internally by the component. That was what I was trying to communicate. I use the state to track whether or not the user has provided those props - thus wants to manage the expanded state of the toggle group themself. 

In the react docs I do try to provide a write up explaining some of that. Maybe that clears it up some? I could add more of that into the comments of the component code for the sake of the developers in the future.

Reply:
Storing `isToggleManaged` in state seems like an antipattern to me, since it is something derived from props. If those props were to change, the state wouldn't update. Since the component will probably never go from managed to unmanaged or vice versa while it is mounted (I don't see why a consumer would do that) it's probably fine here, but I could see this pattern causing bugs if used elsewhere. Instead maybe putting that logic in a method and calling `this.isToggleManaged()` wherever you need to check that would be better.

Reply:
oh I see